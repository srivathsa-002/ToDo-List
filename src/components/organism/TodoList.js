import React from "react";
import TodoForm from "../molecule/TodoForm";
import Todo from "../molecule/Todo";
import "../../styles/components/container.css"
import "../../styles/components/widget.css";
import "../../styles/components/option.css";
import NavigationButtons from "../molecule/NavigationButtons";

export default class TodoList extends React.Component {
  state = {
    todos: [],
    todoToShow: "all",
    toggleAllComplete: true
  };

  addTodo = todo => {
    this.setState(state => ({
      todos: [todo, ...state.todos]
    }));
  };

  editTodo = (oldTodo, newValue) => {
    oldTodo.text = newValue;
  }

  toggleComplete = id => {
    this.setState(state => ({
      todos: state.todos.map(todo => {
        if (todo.id === id) {
          return {
            ...todo,
            complete: !todo.complete
          };
        } else {
          return todo;
        }
      })
    }));
  };

  updateTodoToShow = str => {
    this.setState({
      todoToShow: str
    });
  };

  handleDeleteTodo = id => {
    this.setState(state => ({
      todos: state.todos.filter(todo => todo.id !== id)
    }));
  };

  removeAllTodosThatAreComplete = () => {
    this.setState(state => ({
      todos: state.todos.filter(todo => !todo.complete)
    }));
  };

  render() {
    let todos = [];

    if (this.state.todoToShow === "all") {
      todos = this.state.todos;
    } else if (this.state.todoToShow === "active") {
      todos = this.state.todos.filter(todo => !todo.complete);
    } else if (this.state.todoToShow === "complete") {
      todos = this.state.todos.filter(todo => todo.complete);
    }

    return (
      <div data-testid="todoList">
        <div className="row">
          <div className="column" style={{ float: "left", width: "20vw" }}>
            <NavigationButtons badgeContent={this.state.todos.filter(todo => !todo.complete).length} all={() => this.updateTodoToShow("all")} active={() => this.updateTodoToShow("active")} complete={() => this.updateTodoToShow("complete")}/>
            {/* <div className="button-container">
              <div className="three-buttons">
                <div className="hoverable1" onClick={() => this.updateTodoToShow("all")} val="All">
                  <AllIcon />
                  {/* <ListIcon />*/}
                  {/* {" "}All 
                </div>
                <hr />
                <div className="hoverable2" onClick={() => this.updateTodoToShow("active")} val="Active">
                  <ActiveIcon />
                  <AccessibilityNewIcon />
                  <Badge badgeContent={this.state.todos.filter(todo => !todo.complete).length} color="primary">
                    {" "}Active
                  </Badge>
                </div>
                <hr />
                <div className="hoverable3" onClick={() => this.updateTodoToShow("complete")} val="Complete">
                  <CompleteIcon /> */}
                  {/* <PlaylistAddCheckIcon /> */}
                  {/* {" "}Completed
                  </div>
              </div>
            </div> */}
          </div>
          <div className="column" style={{ width: "77vw", float: "left" }}>
            <div className="container" style={{ maxWidth: "100vw" }}>
              <TodoForm onSubmit={this.addTodo} />
              <div className="widget">
                {todos.map((todo) => (
                  <Todo
                    key={todo.id}
                    toggleComplete={() => this.toggleComplete(todo.id)}
                    onDelete={() => this.handleDeleteTodo(todo.id)}
                    todo={todo}
                    editTodo={this.editTodo}
                  />
                ))}
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
