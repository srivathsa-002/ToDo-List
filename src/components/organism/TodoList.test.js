import React from "react";
import {render,fireEvent} from "@testing-library/react";
import TodoList from "./TodoList";
import "@testing-library/jest-dom/extend-expect";
import {create} from "react-test-renderer";

describe("TodoList testing", () => {
    test("Matches the snapshot", () => {
      const input = create(<TodoList />);
      expect(input.toJSON()).toMatchSnapshot();
    });
});

it("renders correctly", () => {
    const{getByTestId} = render(<TodoList />)

    expect(getByTestId("todoList")).toBeInTheDocument()
})