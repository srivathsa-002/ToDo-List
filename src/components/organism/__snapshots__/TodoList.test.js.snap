// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList testing Matches the snapshot 1`] = `
<div
  data-testid="todoList"
>
  <div
    className="row"
  >
    <div
      className="column"
      style={
        Object {
          "float": "left",
          "width": "20vw",
        }
      }
    >
      <div
        className="button-container"
        data-testid="navigationButtons"
      >
        <div
          className="three-buttons"
        >
          <div
            className="hoverable1"
            onClick={[Function]}
            val="All"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              data-testid="all"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 13h2v-2H3v2zm0 4h2v-2H3v2zm0-8h2V7H3v2zm4 4h14v-2H7v2zm0 4h14v-2H7v2zM7 7v2h14V7H7z"
              />
            </svg>
             
            All
          </div>
          <hr />
          <div
            className="hoverable2"
            onClick={[Function]}
            val="Active"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              data-testid="active"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20.5 6c-2.61.7-5.67 1-8.5 1s-5.89-.3-8.5-1L3 8c1.86.5 4 .83 6 1v13h2v-6h2v6h2V9c2-.17 4.14-.5 6-1l-.5-2zM12 6c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
              />
            </svg>
            <span
              className="MuiBadge-root"
            >
               
              Active
              <span
                className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary MuiBadge-invisible"
              >
                0
              </span>
            </span>
          </div>
          <hr />
          <div
            className="hoverable3"
            onClick={[Function]}
            val="Complete"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              data-testid="complete"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 10H2v2h12v-2zm0-4H2v2h12V6zM2 16h8v-2H2v2zm19.5-4.5L23 13l-6.99 7-4.51-4.5L13 14l3.01 3 5.49-5.5z"
              />
            </svg>
             
            Completed
          </div>
        </div>
      </div>
    </div>
    <div
      className="column"
      style={
        Object {
          "float": "left",
          "width": "77vw",
        }
      }
    >
      <div
        className="container"
        style={
          Object {
            "maxWidth": "100vw",
          }
        }
      >
        <form
          className="widget-header"
          data-testid="form"
          onSubmit={[Function]}
        >
          <input
            className="widget-header-input"
            data-testid="input-field"
            name="text"
            onChange={[Function]}
            placeholder="Todo..."
            value=""
          />
          <div
            className="Grid-container"
            onClick={[Function]}
          >
            <div
              className="MuiGrid-root MuiGrid-container"
            >
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  data-testid="addTask"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
                   
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
                style={
                  Object {
                    "paddingLeft": "5px",
                    "paddingTop": "1px",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "paddingLeft": "5px",
                    }
                  }
                >
                  Add task
                </span>
              </div>
            </div>
          </div>
        </form>
        <div
          className="widget"
        />
      </div>
    </div>
  </div>
</div>
`;
